core/src/main/java/org/infinispan/configuration/parsing/ParseUtils.java

```diff
@@ -322,11 +322,33 @@ public static String requireAttributeProperty(final XMLStreamReader reader, int
         }
     }
 
-    public static void ignoreAttribute(XMLExtendedStreamReader reader, Enum<?> attribute) {
-       CONFIG.ignoreXmlAttribute(attribute, reader.getLocation().getLineNumber(), reader.getLocation().getColumnNumber());
+    public static void ignoreAttribute(XMLExtendedStreamReader reader, String attributeName) {
+       CONFIG.ignoreXmlAttribute(attributeName, reader.getLocation().getLineNumber(), reader.getLocation().getColumnNumber());
+    }
+
+    public static void ignoreAttribute(XMLExtendedStreamReader reader, int attributeIndex) {
+        ignoreAttribute(reader, reader.getAttributeName(attributeIndex).getLocalPart());
     }
 
     public static void ignoreElement(XMLExtendedStreamReader reader, Enum<?> element) {
        CONFIG.ignoreXmlElement(element, reader.getLocation().getLineNumber(), reader.getLocation().getColumnNumber());
     }
+
+    public static XMLStreamException elementRemoved(XMLExtendedStreamReader reader, String newElementName) {
+        if (newElementName != null) {
+            return CONFIG.elementRemovedUseOther(reader.getLocalName(), newElementName, reader.getLocation());
+        } else {
+            return CONFIG.elementRemoved(reader.getLocalName(), reader.getLocation());
+        }
+    }
+
+    public static XMLStreamException attributeRemoved(XMLExtendedStreamReader reader, int attributeIndex,
```

Comment:
I don't like the null attribute. why not having two methods ?

Reply:
Lazyness :)